From 04ef9a7ff789aeda4e2e80ae6bc70beb80507d6b Mon Sep 17 00:00:00 2001
From: Rafael Falcao <rafael.vieirafalcao@windriver.com>
Date: Mon, 18 Jul 2022 09:42:01 -0300
Subject: [PATCH] Update openstack Ingress for networking api v1

This change patches the correct apiVersion for Ingress resources created
by the openstack ingress chart to work correctly when stx-openstack is
applied on stx with kubernetes 1.22+ running, fixing the problem
described in [1].

Same applies to spec changes that need to be done according to
io.k8s.api.networking.v1.IngressBackend documentation.

It could be done by upversioning openstack-helm-infra to its latest
commit (currently 1147988b8eba6ab7d1e7af262843f641be6657ff) but this
upversion has a different series of complications that are being
discussed in [2] 

[1] https://bugs.launchpad.net/starlingx/+bug/1980397
[2] https://review.opendev.org/c/starlingx/openstack-armada-app/+/848336

Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Co-authored-by: Rafael Falcao <rafael.vieirafalcao@windriver.com>
[ upversioned openstack-helm-infra base commit ]
Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
[ Upversioned openstack-helm-infra base commit to Caracal ]
Signed-off-by: Daniel Caires <DanielMarques.Caires@windriver.com>
Change-Id: I0bbecc097fdafdf5ebbc3a164b80ba903b5623f2
---
 mariadb/templates/deployment-ingress.yaml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mariadb/templates/deployment-ingress.yaml b/mariadb/templates/deployment-ingress.yaml
index 6fbf3389..ba0d64c9 100644
--- a/mariadb/templates/deployment-ingress.yaml
+++ b/mariadb/templates/deployment-ingress.yaml
@@ -21,7 +21,7 @@ limitations under the License.
 {{- if .Values.manifests.deployment_ingress }}
 {{- $envAll := . }}
 
-{{- $ingressClass := printf "%s-%s" .deployment_name "mariadb-ingress" }}
+{{- $ingressClass := printf "%s" "mariadb-ingress" }}
 
 {{- $serviceAccountName := printf "%s-%s" .deployment_name "ingress" }}
 {{ tuple $envAll "ingress" $serviceAccountName | include "helm-toolkit.snippets.kubernetes_pod_rbac_serviceaccount" }}
@@ -171,7 +171,7 @@ rules:
   - apiGroups:
       - ""
     resourceNames:
-      - {{ printf "%s-%s" .deployment_name $ingressClass | quote }}
+      - {{ printf "%s" .Release.Name | quote }}
     resources:
       - configmaps
     verbs:
-- 
2.34.1
